src/Stratis.Bitcoin/P2P/Peer/ConnectedPeersHelper.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.P2P.Peer
+{
+    /// <summary>Maintains a list of connected peers and ensures their proper disposal.</summary>
+    public class ConnectedPeersHelper : IDisposable
```

Comment:
>I think we can discuss this component a bit further, to my understanding this is just sort of a bridge between a disconnect event on a peer and the call to dispose that peer.

Yes, correct. 

>this could possibly just live in side the peer itself why does it need to be part of the manager?
Because that way we won't be able to implement IDisposable design properly which suggests that by the end of Dispose() call all threads used by the component should be completed. Instead we would have something like OnDisposeCompleted callback and when Dispose() call is finished component won't actually be disposed.